lesson4/README.md

```diff
@@ -1,9 +1,93 @@
+
+The way you store data is using key-value pairs. There are two ways to create/access a SharedPreferences file.
+
+1. [`getSharedPreferences()`](https://developer.android.com/reference/android/content/Context.html#getSharedPreferences(java.lang.String, int)): You can create and access multiple shared preferences file. You can access it by calling the method with the name of the file using the activity.
+    ```java
+    SharedPreferences sharedPref = getActivity().getSharedPreferences(
+            getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+    ```
+
+2. [`getPreferences()`](https://developer.android.com/reference/android/app/Activity.html#getPreferences(int)): Here you are pretty much doing the same thing except you don't need to mention the file name.
+    ```java
+    SharedPreferences sharedPref = getActivity().getPreferences(Context.MODE_PRIVATE);
+    ```
+
+So, whenever a session of the app gets created (inside `onCreate()`), you should read the shared preference file and when the app gets closed (inside `onStop()`), changes made (for ex. the backgroud color is changed to blue) should be written to the file.
+
+```java
+// To read
+SharedPreferences sharedPref = getActivity().getPreferences(Context.MODE_PRIVATE);
+// Default value for when the file is unavilable
+String defaultValue = getResources().getString(R.string.saved_background_default);
+String background = sharedPref.getString(getString(R.string.saved_background), defaultValue);
+
+// To write
+SharedPreferences sharedPref = getActivity().getPreferences(Context.MODE_PRIVATE);
+SharedPreferences.Editor editor = sharedPref.edit();
+editor.putString(getString(R.string.saved_background), currentBackground);
+editor.commit();
+```
+While writing, notice that we gave it the key (`getString(R.string.saved_background)`) and the corresponding value (`currentBackground`).
 
 ### SQL
 
-## App Life Cycle
\ No newline at end of file
+While implementing a SQLite database in Android, you almost always need to make a schema. The schema basically tells the database what columns are needed. This [page](https://developer.android.com/training/basics/data-storage/databases.html) has all the information you might need to add a SQLite database to your app. 
+
+## App Life Cycle
+
+![](https://developer.android.com/images/training/basics/basic-lifecycle.png)
+
+When you create an new project in Android Studio, you will see a lot of boilerplate. If you are new to Android dev, you probably wouldn't have gone beyond using `onCreate()` and `onDestroy()`. But as you advance, you will realize the need for other methods. Other that the create and destroy methods, we'll use `onPause()` and `onResume`.
+
+Read through each of the lessons listed in [this page](https://developer.android.com/training/basics/activity-lifecycle/index.html).
```

Comment:
that'd be great! also leave the link to the full docs so they can read it
if they want :)

On Sun, Jul 10, 2016 at 2:18 PM Shruti Iyer <notifications@github.com>
wrote:

> In lesson4/README.md
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/10#discussion_r70186220>
> :
>
> > \ No newline at end of file
> > +While implementing a SQLite database in Android, you almost always need to make a schema. The schema basically tells the database what columns are needed. This [page](https://developer.android.com/training/basics/data-storage/databases.html) has all the information you might need to add a SQLite database to your app.
> > +
> > +## App Life Cycle
> > +
> > +![](https://developer.android.com/images/training/basics/basic-lifecycle.png)
> > +
> > +When you create an new project in Android Studio, you will see a lot of boilerplate. If you are new to Android dev, you probably wouldn't have gone beyond using `onCreate()` and `onDestroy()`. But as you advance, you will realize the need for other methods. Other that the create and destroy methods, we'll use `onPause()` and `onResume`.
> > +
> > +Read through each of the lessons listed in [this page](https://developer.android.com/training/basics/activity-lifecycle/index.html).
>
> I can add four sections here, each summarizing those four lessons.
> Thoughts?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/10/files/2c381ad1c8554bbe72318e0c4fbd04c0f3757d2e#r70186220>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIhRRQbxVoFc8Ox8rAQRuBK2HBBni7N2ks5qUTd7gaJpZM4JGtn->
> .
>